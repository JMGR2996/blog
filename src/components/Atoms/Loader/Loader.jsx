import React from 'react'
import './Loader.scss'

/**
 * This is a react component that represents a spinner to show while loading external resources.
 */

const Loader = () => (
  <div className='spinner'>
    <div className='dot1' />
    <div className='dot2' />
  </div>
)

export default Loader
