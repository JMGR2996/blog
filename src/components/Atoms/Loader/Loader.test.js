import React from 'react'
import { mount, configure } from 'enzyme'
import Loader from './Loader'
import Adapter from 'enzyme-adapter-react-16'
configure({ adapter: new Adapter() })

describe('Render Spinner', () => {
  it('render correctly Spinner component', () => {
    const LoaderComponent = mount(<Loader />)
    expect(LoaderComponent).toMatchSnapshot()
  })
})
