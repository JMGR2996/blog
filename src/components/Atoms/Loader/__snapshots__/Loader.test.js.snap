// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Spinner render correctly Spinner component 1`] = `
<Loader>
  <div
    className="spinner"
  >
    <div
      className="dot1"
    />
    <div
      className="dot2"
    />
  </div>
</Loader>
`;
